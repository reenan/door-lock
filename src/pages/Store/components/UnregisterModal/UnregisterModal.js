import React from 'react'

import {
  Modal,
  Button,
  Dimmer,
  Loader,
} from 'semantic-ui-react'

export default ({
  isOpen,
  close,
  unregister,
  loading,
}) => (
  <Modal size='small' open={isOpen} onClose={close}>

    <Modal.Header>Unregister</Modal.Header>
    <Modal.Content>
      <p>
        Do you really want to unregister your store? This action is
        irreversible and all your data will be lost.
      </p>

      <Dimmer active={loading}>
        <Loader/>
      </Dimmer>

    </Modal.Content>

    <Modal.Actions>
      <Button content='Cancel' onClick={close} />
      <Button negative content='Unregister' onClick={unregister} />
    </Modal.Actions>

  </Modal>
)
