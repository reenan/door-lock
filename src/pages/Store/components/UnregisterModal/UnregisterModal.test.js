import React from 'react'
import EnzymeToJson from 'enzyme-to-json'
import { mount } from 'enzyme'

import UnregisterModal from './UnregisterModal'

it('renders without crashing', () => {
  const component = mount(
    <UnregisterModal
      isOpen={false}
      loading={false}
      unregister={() => {}}
      close={() => {}}
    />
  )

  expect(EnzymeToJson(component)).toMatchSnapshot()
})
