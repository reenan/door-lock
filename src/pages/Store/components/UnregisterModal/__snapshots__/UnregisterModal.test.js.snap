// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<_default
  close={[Function]}
  isOpen={false}
  loading={false}
  unregister={[Function]}
>
  <Modal
    centered={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    eventPool="Modal"
    onClose={[Function]}
    open={false}
    size="small"
  >
    <Portal
      closeOnDocumentClick={false}
      closeOnEscape={true}
      eventPool="Modal"
      mountNode={<body />}
      onClose={[Function]}
      onMount={[Function]}
      onOpen={[Function]}
      onUnmount={[Function]}
      open={false}
      openOnTriggerClick={true}
    />
  </Modal>
</_default>
`;
